composer.json

```diff
@@ -20,15 +20,24 @@
             "homepage": "https://github.com/friendsofsymfony/FOSHttpCache/contributors"
         }
     ],
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "https://github.com/ddeboer/guzzle6-adapter"
+        }
+    ],
     "require": {
-        "php": ">=5.3.3",
-        "guzzle/guzzle": "~3.8",
+        "php": ">=5.4.8",
         "symfony/event-dispatcher": "~2.3",
-        "symfony/options-resolver": "~2.3"
+        "symfony/options-resolver": "~2.3",
+        "php-http/adapter": "dev-internal_separation",
+        "guzzlehttp/psr7": "~1.0"
```

Comment:
is the request factory in php-http generic enough that we can not require a specific implementation? can it somehow autodiscover what implementation is available? or do we need one concrete implementation so that we can create requests if no request factory is injected to the client?